datafusion/physical-expr/src/expressions/datetime.rs

```diff
@@ -120,16 +120,21 @@ impl PhysicalExpr for DateTimeIntervalExpr {
             (ColumnarValue::Array(array_lhs), ColumnarValue::Scalar(array_rhs)) => {
                 resolve_temporal_op_scalar(&array_lhs, sign, &array_rhs)
             }
+            // This function evaluates operations between a scalar value and an array of temporal
+            // values. One example is calculating the duration between a scalar timestamp and an
+            // array of timestamps (i.e. `now() - some_column`).
+            (ColumnarValue::Scalar(scalar_lhs), ColumnarValue::Array(array_rhs)) => {
+                let array_lhs = scalar_lhs.to_array_of_size(array_rhs.len());
+                Ok(ColumnarValue::Array(resolve_temporal_op(
```

Comment:
The purpose of the `resolve_temporal_op` is actually to work with real array types. Otherwise, would we need specialized handlers for scalar types? I think this case can be handled in two ways:
1) `resolve_temporal_op_scalar` is extended with a commute parameter. This pattern is applied in `impl_op_arithmetic`. The same functions should be able to handle Scalar op Array and Array op Scalar.
2) A similar resolve_temporal_op_scalar function can be added with replaced parameters of Array and Scalar.
Thank you for working on this issue. I can assist further if needed.

Reply:
@berkaysynnada  are you suggesting we should change this PR? Or can it be merged as is?

I am not quite sure what action your comment is suggesting

Reply:
> @berkaysynnada are you suggesting we should change this PR? Or can it be merged as is?
> 
> I am not quite sure what action your comment is suggesting

There is an operation creating an array from the scalar value. My suggestions intended to prevent this. I think we can merge this PR, I can update that part with a quick PR in the following days.